stencil-library/src/components/table/table.tsx

```diff
@@ -93,8 +93,8 @@ export class Table {
             <tr class="table-light" part='table-head-row'>
               {
                 this.columnData?.map((column) =>
-                  <th part="table-head-cell" scope="col">
-                    {column}
+                  <th part="table-head-cell" scope="col" title={Array.isArray(column) ? column[1] : ''}>
```

Comment:
As a suggestion, not necessary, but we could destruct the array like: 
const [title, content] = column;

Just to have a variable name for column[0] and column[1]